gen/solid.hoon

```diff
@@ -10,59 +10,85 @@
+=/  arvo-src      .^(@t %cx (weld arvo-path /hoon))
+=/  arvo-formula
+  =;  res
+    ?:(?=(%& -.res) p.res (mean (flop p.res)))
+  %-  mule  |.
+  ~&  %solid-loaded
+  =/  compiler-hoon  (rain compiler-path compiler-src)
+  ?.  dub
+    ::  compile arvo against hoon, with our current compiler
+    ::
+    =/  whole-hoon=hoon
+      [%tsbn compiler-hoon [%tsbn [%$ 7] (rain arvo-path arvo-src)]]
     ~&  %solid-parsed
-    =+  one=(~(mint ut %noun) %noun gen)
-    ~&  %solid-compiled
-    ?.  dub
-      =+  two=(~(mint ut p.one) %noun (rain arp rax))
-      ~&  %solid-arvo
-      [7 q.one q.two]
-    =+  zax=(cat 3 '=>  ' (cat 3 txt (cat 3 '  ' rax))) 
-    ~&  %solid-double-loading
-    =+  all=.*(0 q.one)
-    ~&  %solid-double-loaded
+    =/  whole-formula  q:(~(mint ut %noun) %noun whole-hoon)
+    ~&  %solid-arvo
+    whole-formula
+  ::  compile arvo against hoon, with a freshly compiled hoon (via +ride)
+  ::
+  ~&  %solid-parsed
+  =/  compiler-formula  q:(~(mint ut %noun) %noun compiler-hoon)
+  ~&  %solid-compiled
+  =/  whole-src
+    (rap 3 ['=>  ' compiler-src '=>  +7  ' arvo-src ~])
+  ~&  %solid-double-loaded
+  =/  whole-formula
     =<  +
-    .*(all [9 2 [0 2] [1 %noun zax] [0 7]])
+    .*  0
+    :+  %7
+      compiler-formula
+    [%9 2 %10 [6 %1 %noun whole-src] [%0 1]]
+  ~&  %solid-double-compiled
+  whole-formula
 ::
-~&  [%solid-kernel `@ux`(mug ken)]
-:-  ken
-=+  all=.*(0 ken)
-=+  ^=  vay  ^-  (list {p/@tas q/path})
-    :~  [%$ /zuse]
-        [%f /vane/ford]
-        [%b /vane/behn]
-        [%d /vane/dill]
-        [%a /vane/ames]
-        [%c /vane/clay]
-        [%g /vane/gall]
-        [%e /vane/eyre]
-        [%j /vane/jael]
-    ==
-|-  ^+  all
-?~  vay  all
-=+  pax=(weld top.arg q.i.vay)
-=+  txt=.^(@ %cx (weld pax `path`[%hoon ~]))
-=+  sam=[now `ovum`[[%gold ~] [%veer p.i.vay pax txt]]]
-~&  [%solid-veer i.vay]
-::  +poke at +10
+~&  [%solid-kernel `@ux`(mug arvo-formula)]
+::  module-ova: vane load operations.
+::
+=/  module-ova=(list ovum)
+    |^  :~  (vent %$ /zuse)
+            (vent %a /vane/ames)
+            (vent %b /vane/behn)
+            (vent %c /vane/clay)
+            (vent %d /vane/dill)
+            (vent %e /vane/eyre)
+            (vent %f /vane/ford)
+            (vent %g /vane/gall)
+            (vent %j /vane/jael)
+        ==
+    ++  vent
+      |=  [abr=term den=path]
+      ^-  ovum
+      =/  pax  (weld sys den)
+      =/  txt  .^(@ %cx (weld pax /hoon))
+      [[%vane den] [%veer abr pax txt]]
+    --
+::  arvo with vanes installed
+::
+=/  installed
+  =<  q
+  %^    spin
+      module-ova
+    .*(0 arvo-formula)
+  |=  [ovo=ovum ken=*]
+  [~ .*(ken [%9 2 %10 [6 %1 now ovo] %0 1])]
+::  produce both the arvo-formula (for jet registration)
+::  and a kernel with the vanes already installed
 ::
-=+  gat=.*(all .*(all [0 10]))
-=+  nex=+:.*([-.gat [sam +>.gat]] -.gat)
-$(vay t.vay, all nex)
-
+[arvo-formula .*(installed [%0 7])]
```

Comment:
I assume there's a good reason, but why do we grab the context from the 'installed' kernel?

Reply:
Is it to dig the kernel core out of the +poke gate wrapper?

Reply:
`installed` is the formal Arvo interface, ie the anonymous poke wrapper gate, but `release-candidate` expects to be the structural Arvo interface.

Reply:
There should probably be a comment here.